packages/dashboard/src/vaadin-dashboard-helpers.js

```diff
@@ -129,3 +123,28 @@ export function findAncestorInstance(node, baseClass) {
   }
   return null;
 }
+
+/**
+ * Walks up the DOM tree starting from `node`, returning the first ancestor which is an instance of the given `baseClass`.
+ *
+ * @param {Node} node - starting node
+ * @param {Function} baseClass - constructor, e.g. `Dashboard`
+ * @returns {HTMLElement | null}
+ */
+export function findAncestorInstance(node, baseClass) {
+  return __findFilteredAncestorInstance(node, (el) => {
+    return el instanceof baseClass;
+  });
+}
+
+/**
+ * Walks up the DOM tree starting from `node`, returning the first ancestor which extends 'DashboardLayoutMixin'.
+ *
+ * @param {Node} node - starting node
+ * @returns {HTMLElement | null}
+ */
+export function getParentLayout(node) {
+  return __findFilteredAncestorInstance(node, (el) => {
+    return el.constructor && el.constructor.toString().includes('DashboardLayoutMixin');
```

Comment:
This probably won't work in production where function and class names get minified. Did a quick test in a Flow project and "DashboardLayoutMixin" does not show up in any generated chunks. Since there are only two types of elements to test for, we could use the `is` getter to check for the tag name: https://github.com/vaadin/web-components/blob/8ecd3138096ea5aa01580d482ce0e5072c7e9e0e/packages/dashboard/src/vaadin-dashboard.js#L110-L112

Reply:
Updated the helper to use `is`.

Reply:
@web-padawan Since you noted this in the other PR, should we prefer `localName` over `is`?

Reply:
Custom elements (like widgets) that have different tag names do not seem to update correctly when we use `localName` instead.

Reply:
Updated to use the private property `__hasVaadinDashboardLayoutMixin`, similar to `_hasVaadinItemMixin`.